docs/06-concepts/11-authentication/05-custom-overrides.md

```diff
@@ -42,6 +46,36 @@ List<String> scopes = extractScopes(token);
 Set<Scope> userScopes = scopes.map((scope) => Scope(scope)).toSet();
 ```
 
+### Handling revoked authentication 
+
+When a user's authentication is revoked, the server must be promptly notified to maintain secure access control. The `authenticationRevoked` event plays a crucial role in closing method streams that the user is no longer authorized to access. To trigger this event, invoke the `session.messages.authenticationRevoked` method.
+
+```dart
+await session.messages.authenticationRevoked(
+  userId,
+  RevokedAuthenticationScope(
+    scopes: removedScopes,
+  ),
```

Comment:
Question (for my understanding): From an API point of view, why are we sending the diff (`removedScopes`) in the event instead of the actual scopes that the user has? (For some reason I didn't think about this when I reviewed the actual PR) e.g.
```
await session.messages.authenticationRevoked(
  userId,
  RevokedAuthenticationScope(
    newScopes: newScopes,
  )
```
I would imagine in most use cases, it would be more straight forward to pass in the current state of things?

Reply:
No, we send the revoked scopes, so the example is correct. To make this even clearer, removed scopes should probably be called `revokedScoped` in the example.

The purpose of this is to make it as easy and declarative as possible to post only what is revoked.

Reply:
Thanks!